app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -194,5 +184,27 @@ def user_locale
     def otp_rate_limiter
       @_otp_rate_limited ||= OtpRateLimiter.new(phone: phone_to_deliver_to, user: current_user)
     end
+
+    def redirect_on_nothing_enabled
+      redirect_to two_factor_options_url
+    end
+
+    def redirect_on_phone
+      return unless phone_enabled?
+      validate_otp_delivery_preference_and_send_code
+      true
+    end
+
+    def redirect_on_non_phone
+      if current_user.piv_cac_enabled?
+        redirect_to login_two_factor_piv_cac_url
+      elsif current_user.webauthn_enabled?
+        redirect_to login_two_factor_webauthn_url
+      elsif current_user.totp_enabled?
+        redirect_to login_two_factor_authenticator_url
+      else
+        false
```

Comment:
Style: this might not be needed. If no condition is true, then none of the redirects will be called, and `nil` will be returned.